lib/commands/app/rename.ts

```diff
@@ -68,38 +71,22 @@ export default class AppRenameCmd extends Command {
 	public async run() {
 		const { args: params } = this.parse<FlagsDef, ArgsDef>(AppRenameCmd);
 
-		const { ExpectedError, instanceOf } = await import('../../errors');
-		const { getApplication } = await import('../../utils/sdk');
+		const { validateApplicationName } = await import('../../utils/validation');
+		const { ExpectedError } = await import('../../errors');
+
 		const balena = getBalenaSdk();
 
-		// Get app
-		let app;
-		try {
-			app = await getApplication(balena, params.name, {
-				$expand: {
-					application_type: {
-						$select: ['is_legacy'],
-					},
-				},
-			});
-		} catch (e) {
-			const { BalenaApplicationNotFound } = await import('balena-errors');
-			if (instanceOf(e, BalenaApplicationNotFound)) {
-				throw new ExpectedError(`Application ${params.name} not found.`);
-			} else {
-				throw e;
-			}
-		}
+		// Disambiguate target application (if nameOrSlug is a number, it could either be an ID or a numerical name)
+		const { getApplication } = await import('../../utils/sdk');
+		const application = await getApplication(balena, params.nameOrSlug);
 
-		// Check app supports renaming
-		const appType = (app.application_type as ApplicationType[])?.[0];
-		if (appType.is_legacy) {
+		if (!application) {
 			throw new ExpectedError(
-				`Application ${params.name} is of 'legacy' type, and cannot be renamed.`,
```

Comment:
Legacy applications are still supported by balena as far as I am aware. I think we need this logic. (I reckon this PR proposes to delete the logic altogether.)

Reply:
If I remember correctly from a conversation with Thodoris, this is handled by the SDK (I'll check).

Reply:
@thgreasi I think you mentioned that the SDK application.rename command handles legacy app checking (though not  sure I remember correctly).  Do you know if this is the case?

Reply:
@pdcastro Ok, I've put this check back.  If it turns out to be redundant we can remove it.

Reply:
I don't see any special check in the SDK, but I'm pretty sure the API blocks it.